mitodl/open-discussions
# Field edit page with preliminary appearance tab

#### Pre-Flight checklist


#### What are the relevant tickets?
Partially addresses #3632 

#### What's this PR do?
Adds a form for editing field title and description, but not banner and avatar.  These will be added in a separate PR.
Adds a tabbed edit page with 3 tabs (basic, appearance, moderators).  All but appearance tab are just placeholders right now.
Creates a separate component for field avatar
Adds `is_moderator` to the API response to determine if the current user can access the field edit page/forms.

#### How should this be manually tested?
- As an admin, go to the url `/infinite/fields/<your_field_name>/manage/` and verify that you see 3 tabs.
- Go to the "Appearance" tab and verify the title and description input fields are prepopulated with current values.
- Make both blank and verify you see validation errors
- Enter new values and submit.  You should be redirected back to the field page and see your changes there.
- As a non-admin/non-moderator your field, return to the edit page.  You should see a message that you're not allowed to access the page.

#### Screenshots (if appropriate)

<img width="776" alt="Screen Shot 2022-08-02 at 10 37 44 AM" src="https://user-images.githubusercontent.com/187676/182401498-03c90051-1fc3-49d1-9e0b-e3edef6d7889.png">
